packages/sdk/src/cross-chain-messenger.ts

```diff
@@ -228,6 +229,27 @@ export class CrossChainMessenger {
     }
   }
 
+  /**
+   * Uses portal version to determine if the messenger is using fpac contracts. Better not to cache
+   * this value as it will change during the fpac upgrade and we want clients to automatically
+   * begin using the new logic without throwing any errors.
+   *
+   * @returns Whether or not the messenger is using fpac contracts.
+   */
+  public async fpac(): Promise<boolean> {
+    if (
+      this.contracts.l1.OptimismPortal.address === ethers.constants.AddressZero
+    ) {
+      // Some networks don't have a portal, so we can't use it to determine if we're using fpac.
```

Comment:
```suggestion
      // Some networks don't have a portal, so we can't use it to determine if we're using fpac.
```
nit: Could be helpful to future readers to provide an example of a network that doesn't have a portal? Just a thought don't feel like it's necessary

Reply:
Good point, I feel like we can remove this

Reply:
Nvm I remember why I did this, it's actually because the tests sometimes don't deploy a portal. Changed the comment to reflect this.